api/app/Models/User.php

```diff
@@ -1124,12 +1135,12 @@ public function scopeAuthorizedToViewBasicInfo(Builder $query): void
     }
 
     /**
-     * Determine if the user has verified their email address.
+     * Determine if the user has verified their contact email address.
      * Part of the MustVerifyEmail contract.
      *
      * @return bool
      */
-    public function hasVerifiedEmail()
+    public function hasVerifiedContactEmail()
```

Comment:
This model now has a mix of pairs of functions, one for each emaail type, and single functions that accept an emailType parameter.  It would be nice if this were more consistent.  On one hand, the pairs are more convenient and conform to the MustVerifyEmail contract better.  On the other hand, you've made good use of a single `getEmailForVerification` function.  Any ideas?

Reply:
I definitely prefer doing things just one way and merged the last pair of functions in [Merge hasVerifiedEmail functions](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11470/commits/70dc9a5dace693728e5dbc7f686d2ed88f824252).
I don't really know the details of the MustVerifyEmail contract though and would be fine with swapping everything to paired functions if you think that would work better.

Reply:
I like it!

You can see the contract in api/vendor/laravel/framework/src/Illuminate/Contracts/Auth/MustVerifyEmail.php .  We've really moved away from it in this PR but I think that's fine.  It was a bigger deal when the team was a bit nervous about not using the "pure Laravel" way of validation emails.